src/plugins/memdump/memdump.cpp

```diff
@@ -753,6 +753,103 @@ static event_response_t write_virtual_memory_hook_cb(drakvuf_t drakvuf, drakvuf_
     return VMI_EVENT_RESPONSE_NONE;
 }
 
+static event_response_t create_remote_thread_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info) {
+    // First check if NtCreateThreadEx syscall was invoked from CreateRemoteThread. 
+    // In such case the target process should differ from the caller.
+
+    // IN HANDLE ProcessHandle
+    addr_t target_process_handle = drakvuf_get_function_argument(drakvuf, info, 4);
+
+    vmi_pid_t target_process_pid;
+    if (!drakvuf_get_pid_from_handle(drakvuf, info, target_process_handle, &target_process_pid)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve target process pid\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    addr_t caller_process = drakvuf_get_current_process(drakvuf, info);
+    if (!caller_process) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve caller process\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+    vmi_pid_t caller_process_pid;
+    if (!drakvuf_get_process_pid(drakvuf, caller_process, &caller_process_pid)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve caller process pid\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    if (target_process_pid == caller_process_pid) {
```

Comment:
If I'm not mistaken, this could be simplified al ittle bit.

```suggestion
    if (target_process_pid == info->proc_data.pid) {
```